src/cargo/ops/tree/format/parse.rs

```diff
@@ -0,0 +1,97 @@
+use std::iter;
+use std::str;
+
+pub enum RawChunk<'a> {
+    Text(&'a str),
+    Argument(&'a str),
+    Error(&'static str),
+}
+
+pub struct Parser<'a> {
```

Comment:
Could this have a brief comment of the mini-language it's parsing?

Reply:
The parser seems a little more elaborate than what is necessary (as compared to a simple `.replace()` substitution).  Perhaps @sfackler had grander plans for it?

Reply:
I think I just stole most of it from the fmt syntax parser in the standard library.

Reply:
Oh I think it's fine and probably inevitable, string-replacement can't do things like `{{t}}` generating the text `{t}` without doing `{$replacement_for_t}`, and it's all small and containe enough anyway.